draft-ietf-tls-esni.md

```diff
@@ -562,13 +575,14 @@ The ClientEncryptedSNI.encrypted_sni value is then computed using the usual
 TLS 1.3 AEAD:
 
 ~~~~
-    encrypted_sni = AEAD-Encrypt(key, iv, KeyShareClientHello, ClientESNIInner)
+    encrypted_sni = AEAD-Encrypt(key, iv, Hash(ClientHelloPlaintext),
+                                 ClientESNIInner)
 ~~~~
 
-Where KeyShareClientHello is the "extension_data" field of the "key_share"
-extension in a Client Hello (Section 4.2.8 of {{!RFC8446}})). Including
-KeyShareClientHello in the AAD of AEAD-Encrypt binds the ClientEncryptedSNI
-value to the ClientHello and prevents cut-and-paste attacks.
+where ClientHelloPlaintext is the entire ClientHello to be transformed but the
+"server_name" extension being skipped.  Feeding the unencrypted part of the
+ClientHello in the AAD of AEAD-Encrypt binds the encrypted value to the prevents
+cut-and-paste attacks.
 
```

Comment:
binds the encrypted value *and* prevents cut-and-paste attacks.

Reply:
Fixed in c62ee5d.